internal/pkg/flink/internal/store/store.go

```diff
@@ -189,10 +197,36 @@ func (s *Store) waitForPendingStatement(ctx context.Context, statementName strin
 
 	var errorsMsg string
 	if len(capturedErrors) > 0 {
-		errorsMsg = fmt.Sprintf(" Captured retriable errors: %s", strings.Join(capturedErrors, "; "))
+		errorsMsg = fmt.Sprintf("Captured retriable errors: %s", strings.Join(capturedErrors, "; "))
+	}
+
+	return nil, &types.StatementError{
+		Msg: fmt.Sprintf("Statement is still pending after %f seconds.\n If you want to increase the timeout for the client, you can run \"SET table.results-timeout=1200;\" to adjust the maximum timeout in seconds.",
+			timeout.Seconds()),
+		FailureMessage: errorsMsg,
+	}
+}
+
+func (s *Store) getStatusDetail(statementObj flinkgatewayv1alpha1.SqlV1alpha1Statement) string {
+	status := statementObj.GetStatus()
+	phase := types.PHASE(status.GetPhase())
+	if phase != types.FAILED && phase != types.FAILING {
+		return status.GetDetail()
+	}
+
+	if status.GetDetail() != "" {
+		return status.GetDetail()
+	}
+
+	// if the statement is in FAILED or FAILING phase and the status detail field is empty we show the latest exception instead
+	exceptionsResponse, _ := s.client.GetExceptions(s.appOptions.GetOrgResourceId(), s.appOptions.GetEnvId(), statementObj.Spec.GetStatementName())
+	exceptions := exceptionsResponse.GetData()
+	if len(exceptions) < 1 {
+		return status.GetDetail()
```

Comment:
Looks like at this point `status.GetDetail()` would be always empty here, right?

Reply:
Yes, it would be equivalent to just return empty string here! I can change it if you prefer it that way

Reply:
Let's make it empty string to be more explicit about the intent.

Reply:
Works for me! And I updated it yesterday already 👍